### PR TITLE
Anomali threat stream v2 - Added Reliability

### PR DESCRIPTION
**Pull request by yaakovpraisler**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
related: demisto/etc#29347

## Description
Added reliability

## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [x] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py [hunk 1]
```diff
@@ -967,49 +965,73 @@ def main():
...
     command = demisto.command()
     LOG(f'Command being called is {command}')
+
+    params = demisto.params()
+
+    user_name = params.get('username')
+    api_key = params.get('apikey')
+
+    CREDENTIALS['user_name'] = user_name
+    CREDENTIALS['api_key'] = api_key
```
Comment (bakatzir) [inline Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py:975]: shouldn't they be declared as global?
@DeanArbel wdyt?

File: Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py [hunk 2]
```diff
@@ -155,7 +152,7 @@ def build_params(**params):
     return params
 
 
-def get_dbot_context(indicator, threshold, mapping=None):
+def get_dbot_context(client: Client, indicator, threshold, mapping=None):
```
Comment (bakatzir) [inline Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py:155]: this can be a static client func, your call.

**Review by bakatzir — COMMENTED** [review]
The refactoring is worthwhile, good work.

**Review by bakatzir — APPROVED** [review]

**Review by bakatzir — COMMENTED** [review]
The TPB is skipped. ddid you ran he commands locally?
https://github.com/demisto/etc/issues/19182
please try to run it locally/unskip the TPB.

**Comment by yaakovpraisler** [comment]
No instance can be configured at the moment, due to https://github.com/demisto/etc/issues/19182.
So this PR will wait, so the TPB can be unskipped.

**Comment by bakatzir** [comment]
@yaakovpraisler please do not miss this PR. we need it to properly close the issue.

**Comment by bakatzir** [comment]
> @yaakovpraisler please do not miss this PR. we need it to properly close the issue.

any update here?

**Comment by yaakovpraisler** [comment]
> > @yaakovpraisler please do not miss this PR. we need it to properly close the issue.
> 
> any update here?

I have some instance config issues. Trying to fix this.

**Comment by bakatzir** [comment]
@yaakovpraisler let's move forward with this without unskipping the TPB. let me know when I can re review the entire PR.

File: Packs/Anomali_ThreatStream/ReleaseNotes/1_1_1.md [hunk 3]
```diff
@@ -2,4 +2,4 @@
 #### Integrations
 ##### Anomali ThreatStream v2
 - Fixed an issue where the *tags* information was missing from the context outputs of ***ip***, ***domain***, ***file***, ***url***, and ***threatstream-email-reputation*** commands.
-- Updated the Docker image to: *demisto/emoji:1.0.0.18288*
+- Updated the Docker image to: *demisto/emoji:1.0.0.19032*
```
Comment (bakatzir) [inline Packs/Anomali_ThreatStream/ReleaseNotes/1_1_1.md:5]: please revert this file changes.

File: Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py [hunk 4]
```diff
@@ -1013,49 +1011,75 @@ def main():
...
+
+    if DBotScoreReliability.is_valid_type(reliability):
+        reliability = DBotScoreReliability.get_dbot_score_reliability_from_str(reliability)
+    else:
+        Exception("Please provide a valid value for the Source Reliability parameter.")
+
     try:
+
+        client = Client(
+            base_url=server_url + '/api/',
```
Comment (bakatzir) [inline Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py:1035]:
```suggestion
            base_url=f'{server_url}/api/',
```

File: Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py [hunk 5]
```diff
@@ -87,37 +77,60 @@
...
+             Builds and returns dictionary with Indicator, Type, Vendor and Score keys
+             and values from the indicator that will be returned to context.
+        """
+        if not mapping:
+            mapping = DBOT_MAPPING
+        dbot_context = {mapping[k]: v for (k, v) in indicator.items() if k in mapping.keys()}
+        indicator_score = DBOT_SCORE[indicator.get('meta', {}).get('severity', 'low')]
+        # the indicator will be considered as malicious in case it's score is greater or equal to threshold
+        dbot_context['Score'] = 3 if indicator_score >= DBOT_SCORE[threshold] else indicator_score
+        dbot_context['Vendor'] = 'ThreatStream'
```
Comment (bakatzir) [inline Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py:130]: I suggest putting `ThreatStream` in a global var, so you can use it in here, error messages and in the contextpaths you create along the integration commands.

File: Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py [hunk 6]
```diff
@@ -87,37 +77,60 @@
...
+        elif res.status_code in {404}:
+            command = demisto.command()
+            if command in ['threatstream-get-model-description', 'threatstream-get-indicators-by-model',
+                           'threatstream-get-analysis-status', 'threatstream-analysis-report']:
+                # in order to prevent raising en error in case model/indicator/report was not found
+                return {}
+            else:
+                return_error("The resource not found. Check the endpoint.")
+        elif res.status_code not in {200, 201, 202}:
+            return_error(F"Error in API call to ThreatStream {res.status_code} - {res.text}")
```
Comment (bakatzir) [inline Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py:113]: you can raise Exception here, it is caught in the main

File: Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py [hunk 7]
```diff
@@ -87,37 +77,60 @@
...
+            verify=self.use_ssl,
+            params=params,
+            data=data,
+            headers=headers,
+            files=files,
+            json=json
+        )
+        # Handle error responses gracefully
+        if res.status_code in {401}:
+            return_error("Got unauthorized from the server. Check the credentials.")
```
Comment (bakatzir) [inline Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py:103]: you can raise Exception here, it is caught in the main

File: Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py [hunk 8]
```diff
@@ -87,37 +77,60 @@
...
+        elif res.status_code in {404}:
+            command = demisto.command()
+            if command in ['threatstream-get-model-description', 'threatstream-get-indicators-by-model',
+                           'threatstream-get-analysis-status', 'threatstream-analysis-report']:
+                # in order to prevent raising en error in case model/indicator/report was not found
+                return {}
+            else:
+                return_error("The resource not found. Check the endpoint.")
```
Comment (bakatzir) [inline Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py:111]: you can raise Exception here, it is caught in the main

Comment (bakatzir) [reply Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py:111]: also add the `ThreatStream` string on the prefix for clarity

**Review by bakatzir — COMMENTED** [review]
nice job, a few small comments

File: Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py (hunk 4 above)
Comment (yaakovpraisler) [reply Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py]: done.

File: Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py (hunk 5 above)
Comment (yaakovpraisler) [reply Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py]: done.

File: Packs/Anomali_ThreatStream/ReleaseNotes/1_1_1.md (hunk 3 above)
Comment (yaakovpraisler) [reply Packs/Anomali_ThreatStream/ReleaseNotes/1_1_1.md]: done.

**Review by bakatzir — APPROVED** [review]
fails on the known
```
Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.yml: [IN140] - The integration Anomali ThreatStream v2 is currently in skipped. Please add working tests and unskip. Skip comment: Issue 19182

Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.yml: [IN116] - You've added required, the field is 'integrationReliability'
```
will merge.

**Pull request merged**